tests/unit/memefficiency.tcl

```diff
@@ -145,601 +287,323 @@ run_solo {defrag} {
-                    if {$eventname == "active-defrag-cycle"} {
-                        set max_latency $max
-                    }
-                }
-                if {$::verbose} {
-                    puts "frag $frag"
-                    set misses [s active_defrag_misses]
-                    set hits [s active_defrag_hits]
-                    puts "hits: $hits"
-                    puts "misses: $misses"
-                    puts "max latency $max_latency"
-                    puts [r latency latest]
-                    puts [r latency history active-defrag-cycle]
+            perform_defrag_test $title {
+                set rd [valkey_deferring_client]
+                set val [string repeat A 50]
+                for {set j 0} {$j < $n} {incr j} {
+                    $rd xadd k$j * field1 $val field2 $val
                 }
-                assert {$frag < 1.1}
-                # due to high fragmentation, 100hz, and active-defrag-cycle-max set to 75,
-                # we expect max latency to be not much higher than 7.5ms but due to rare slowness threshold is set higher
-                if {!$::no_latency} {
-                    assert {$max_latency <= 30}
+                for {set j 0} {$j < $n} {incr j} { $rd read } ; # Discard replies
+            } fragment {
+                for {set j 0} {$j < $n} {incr j 2} {
+                    $rd del k$j
                 }
+                for {set j 0} {$j < $n} {incr j 2} { $rd read } ; # Discard replies
+                $rd close
             }
-            # verify the data isn't corrupted or changed
-            set newdigest [debug_digest]
-            assert {$digest eq $newdigest}
-            r save ;# saving an rdb iterates over all the data / pointers
-        } {OK}
-
-        test "Active defrag pubsub: $type" {
-            r flushdb sync
-            r config resetstat
-            r config set activedefrag no
-            r config set active-defrag-threshold-lower 5
-            r config set active-defrag-cycle-min 65
-            r config set active-defrag-cycle-max 75
-            r config set active-defrag-ignore-bytes 1500kb
-            r config set maxmemory 0
+        }
+    }
+
+    proc test_pubsub {type} {
+        set title "Active Defrag pubsub: $type"
+        test $title {
+            set n 100000
 
-            # Populate memory with interleaving pubsub-key pattern of same size
-            set n 50000
-            set dummy_channel "[string repeat x 400]"
             set rd [valkey_deferring_client]
-            set rd_pubsub [valkey_deferring_client]
+            set chan [string repeat A 50]
+
+            # https://github.com/valkey-io/valkey/issues/1774
+            # NOTE - pubsub defrag isn't working properly.  This wasn't caught before the test refactor.
+            #  This commented out code should work.  Instead, is substituted code which verifies that
+            #  defrag doesn't break the pubsub, but defrag efficacy is not verified.
```

Comment:
These lines are too long IMHO. Even if we allow longer lines of code, it's easier to read comments that are not that wide (especially on a phone, or laptop with e.g. side-by-side editor and terminal). Are you using emacs and automatic reflow using M-q (fill-paragraph)? I suggest setting fill-column to 80.

Reply:
```suggestion
            # NOTE - pubsub defrag isn't working properly.  This wasn't caught before the
            #  test refactor.  This commented out code should work.  Instead, is substituted
            # code which verifies that defrag doesn't break the pubsub, but defrag efficacy
            # is not verified.
```